src/platforms/esp32/components/avm_builtins/network_driver.c

```diff
@@ -595,7 +608,7 @@ static void start_network(Context *ctx, term pid, term ref, term config)
     }
 
     struct ClientData *data = malloc(sizeof(struct ClientData));
-    if (IS_NULL_PTR(data)) {
+    if (UNLIKELY(IS_NULL_PTR(data))) {
```

Comment:
My understanding is that IS_NULL_PTR already does the optimizations that UNLIKELY does, so the UNLIKELY is not needed.

Reply:
indeed, IS_NULL_PTR is roughly defined as `UNLIKELY(data == NULL)`.

Reply:
Thank you. I had temporarily forgotten that the UNLIKELY was handled in the macro.